src/index.js

```diff
@@ -8,272 +8,359 @@ const _ = require("lodash");
-        );
-        outputMismatchResults.push(
-          `Test output do not match for transformation: ${transformationName}`
-        );
-
-        fs.writeFileSync(
-          `${testOutputDir}/${transformationHandleName}_diff.json`,
-          JSON.stringify(detailedDiff(expectedOutput, apiOutput), null, 2)
-        );
-
-        testOutputFiles.push(
-          `${testOutputDir}/${transformationHandleName}_diff.json`
-        );
-      }
     }
+  }
+}
 
-    // upload artifact
-    if (uploadTestArtifact === "true") {
-      core.info("Uploading test api output...");
-      await artifactClient.uploadArtifact(
-        "transformer-test-results",
-        testOutputFiles,
-        "."
+// Upload the test results to an artifact store.
+async function uploadTestArtifacts(testOutputFiles) {
+  core.info(`Uploading test artifacts`);
+  // upload artifact
+  if (uploadTestArtifact) {
+    core.info("Uploading test api output");
+    const artifactClientResponse = await artifactClient.uploadArtifact(
+      "transformer-test-results",
+      testOutputFiles,
+      "."
+    );
+
+    if (artifactClientResponse.failedItems.length > 0) {
+      throw new Error(
+        `Artifacts upload failed, failedItem: ${JSON.stringify(failedItems)}`
       );
     }
+  }
+}
 
-    if (outputMismatchResults.length > 0) {
-      throw new Error(outputMismatchResults.join(", "));
-    }
+// Publish the transformations and libraries.
+async function publishTransformation(
+  transformationTest,
+  librariesTest,
+  commitId
+) {
+  core.info(`Publishing transformations and libraries`);
+  // publish
+  res = await publish(transformationTest, librariesTest, commitId);
+  core.info(`Publish result: ${JSON.stringify(res.data)}`);
+}
 
-    // test passed
-    core.info("Test Passed!!!");
+function colorize(message, color) {
+  const colors = {
+    reset: "\x1b[0m",
+    red: "\x1b[31m",
+    green: "\x1b[32m",
+    yellow: "\x1b[33m",
+    blue: "\x1b[34m",
+    magenta: "\x1b[35m",
+    cyan: "\x1b[36m",
+    white: "\x1b[37m",
+  };
+
+  return `${colors[color]}${message}${colors.reset}`;
+}
 
-    // publish
-    if (!testOnly) {
-      res = await publish(transformationTest, librariesTest, commitId);
-      core.info(`Publish result: ${JSON.stringify(res.data)}`);
-    }
-  } catch (err) {
-    if (err.response) {
-      core.error(err.response.data);
-      core.error(err.response.status);
-    }
-    core.setFailed(err);
+// Log failed tests
+function logFailedTests(tests) {
+  core.info(colorize("\nFailed Tests:\n", "yellow"));
+  for (const test of tests) {
+    core.info(colorize(`   ID: ${test.id}`, "red"));
+    core.info(colorize(`   Name: ${test.name}`, "red"));
+    core.info(colorize(`   Error: ${JSON.stringify(test.result)}\n`, "red"));
+    core.info("\n" + "=".repeat(40) + "\n"); // Add a line of equal signs between logs
+  }
+}
+
+async function testAndPublish() {
+  const outputMismatchResults = [];
```

Comment:
Who is setting this outputMismatchResults ? I don't see if we are passing it in any function ?!